packages/react-core/src/hooks/useUserAttributes/action.ts

```diff
@@ -0,0 +1,172 @@
+  // structure of `userAttributeKeys`
+  await deleteUserAttributes({ userAttributeKeys } as {
+    userAttributeKeys: [UserAttributeKey, ...UserAttributeKey[]];
+  });
+  const attributes = await fetchUserAttributes();
+  return { ...prevResult, attributes };
+};
+
+const fetchAction = async (
+  prevResult: HandleAttributeActionOutput,
+  _: FetchInput
+): Promise<HandleAttributeActionOutput> => {
+  const attributes = await fetchUserAttributes();
+  return { ...prevResult, attributes };
+};
+
+const sendCodeAction = async (
+  prevResult: HandleAttributeActionOutput,
+  { userAttributeKey }: SendCodeInput
+): Promise<HandleAttributeActionOutput> => {
+  const result = await sendUserAttributeVerificationCode({ userAttributeKey });
+  return {
+    ...prevResult,
+    pendingVerification: [
+      {
+        name: userAttributeKey,
+        codeDeliveryDetails: {
+          destination: result.destination,
+          medium: result.deliveryMedium,
+        },
+      },
+    ],
+  };
+};
+
+const updateAction = async (
+  prevResult: HandleAttributeActionOutput,
+  { userAttributes }: UpdateInput
+): Promise<HandleAttributeActionOutput> => {
+  const updates = await updateUserAttributes({ userAttributes });
+  const attributes = await fetchUserAttributes();
+  if (updates.email && updates.phone_number) {
+    return {
+      ...prevResult,
+      attributes,
+      pendingVerification: [
+        {
+          name: 'email',
+          codeDeliveryDetails: {
+            destination:
+              updates.email.nextStep.codeDeliveryDetails?.destination,
+            medium: updates.email.nextStep.codeDeliveryDetails?.deliveryMedium,
+          },
+        },
+        {
+          name: 'phone_number',
+          codeDeliveryDetails: {
+            destination:
+              updates.phone_number.nextStep.codeDeliveryDetails?.destination,
+            medium:
+              updates.phone_number.nextStep.codeDeliveryDetails?.deliveryMedium,
+          },
+        },
+      ],
+    };
+  } else if (updates.email && !updates.phone_number) {
+    return {
+      ...prevResult,
+      attributes,
+      pendingVerification: [
+        {
+          name: 'email',
+          codeDeliveryDetails: {
+            destination:
+              updates.email.nextStep.codeDeliveryDetails?.destination,
+            medium: updates.email.nextStep.codeDeliveryDetails?.deliveryMedium,
+          },
+        },
+      ],
+    };
+  } else if (!updates.email && updates.phone_number) {
+    return {
+      ...prevResult,
+      attributes,
+      pendingVerification: [
+        {
+          name: 'phone_number',
+          codeDeliveryDetails: {
+            destination:
+              updates.phone_number.nextStep.codeDeliveryDetails?.destination,
+            medium:
+              updates.phone_number.nextStep.codeDeliveryDetails?.deliveryMedium,
+          },
+        },
+      ],
+    };
+  } else {
```

Comment:
We can simplify the conditional logic and reduce complexity here by creating a utility for creating the `pendingVerification` value, e.g.:
```ts
const getVerifiableAttribute = (
  attributes: UpdateUserAttributesOutput,  name: keyof UpdateUserAttributesOutput,
):
  | VerifiableAttribute
  | undefined => {
  const attribute = attributes[name]
  if (
    attribute.nextStep.updateAttributeStep === 'CONFIRM_ATTRIBUTE_WITH_CODE'
  ) {
    const { deliveryMedium: medium, destination } =
      attribute.nextStep.codeDeliveryDetails;
    return { name, codeDeliveryDetails: { medium, destination } };
  }
};

...

const updateAction = async (
  prevResult: HandleAttributeActionOutput,
  { userAttributes }: UpdateInput
): Promise<HandleAttributeActionOutput> => {
  const updates = await updateUserAttributes({ userAttributes });
  const attributes = await fetchUserAttributes();

  const pendingEmail = getVerifiableAttribute(updates, 'email');
  const pendingPhoneNumber = getVerifiableAttribute(updates, 'phone_number');

  const pendingVerification = pendingEmail || pendingPhoneNumber ? [
    ...[(pendingEmail ? pendingEmail  : undefined)], 
    ...[(pendingPhoneNumber ? pendingPhoneNumber  : undefined)], 
  ] : undefined;

  return { ...prevResult, attributes, pendingVerification }
]
```